back/app/models/permissions_field.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: permissions_fields
+#
+#  id              :uuid             not null, primary key
+#  permission_id   :uuid             not null
+#  custom_field_id :uuid
+#  required        :boolean          default(TRUE), not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  field_type      :string           default("custom_field")
+#  enabled         :boolean          default(TRUE), not null
+#  config          :jsonb            not null
```

Comment:
Is there a reason to not model this as a separate table (e.g. with the columns permission_field_id, enabled, field_name)? We've had a lot of headaches in the past with using json fields (validation issues and inconsistent data, dependant destroy and other mechanisms).